### ISSUE TITLE
False Positive [E0133] "Access to union field is unsafe" in assigning value to nested union field

### ISSUE THREAD

**Issue by Maksim20023** [intellij-rust/intellij-rust#10831]
## Environment

* **IntelliJ Rust plugin version:** 0.4.200.5421.232
* **Rust toolchain version:** 1.71.1
* **IDE name and version:** CLion 2023.2.1 Build #CL-232.9559.58, built on August 22, 2023
* **Operating system:** macOS 13.4.1

## Problem description

![Screenshot 2023-08-28 at 17 03 34](https://github.com/intellij-rust/intellij-rust/assets/135699123/c46aa4d6-1231-4d9d-afab-54c635a4947d)

## Steps to reproduce
Create a new rust project and copy-paste the following code:
````rust
#![allow(non_camel_case_types)]
#![allow(non_snake_case)]

pub struct BACnet_Application_Data_Value {
    pub type_: BACnet_Application_Data_Value__bindgen_ty_1,
}


pub union BACnet_Application_Data_Value__bindgen_ty_1 {
    pub Special_Event: BACnet_Special_Event,
}

#[derive(Clone, Copy)]
pub struct BACnet_Special_Event {
    pub period: BACnet_Special_Event__bindgen_ty_1,
}


#[derive(Clone, Copy)]
pub union BACnet_Special_Event__bindgen_ty_1 {
    pub calendarReference: BACnet_Object_Id,
}

#[derive(Clone, Copy)]
pub struct BACnet_Object_Id {
    pub type_: u32,
}

fn main() {
    let mut enc: BACnet_Application_Data_Value = unsafe { std::mem::zeroed() };

    // bug here:
    enc.type_.Special_Event.period.calendarReference.type_ = 1;
}
````
